specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+    var uri = new Uri(args.Uri);
+
+    if (String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) &&
+        !coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = true;
+    }
+    else if (!String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) && 
+        coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = false;
+    }
+}
+```
+
+## Declaring Non-client App Regions
+Non-client regions are HTML elements that are marked with the css style `app-region`.
+* Draggable regions can be declared through the values `drag` or `no-drag`. 
+    * `app-region: drag` will support [draggable region functionality](#description) for the html element.
+    * `app-region: no-drag` will change cursor to I-bar, with text highlighting enabled. 
+    Elements with this style will not support draggable region functionality.
+```html
+<!DOCTYPE html>
+<body>
+    <div style="app-region:drag">Drag Region</div>
+    <div style="app-region:no-drag">No-drag Region</div>
+</body>
+<html>
+```
+
+# Remarks
+If the feature flag (`msWebView2EnableDraggableRegions`) is used to enable the CSS style `app-region`'s
+values `drag` and `no-drag`, draggable region support will remain enabled 
+even if the `IsNonClientRegionSupportEnabled` setting is `FALSE`. 
+* Note: The feature flag is experimental and should not be used in production.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+```cpp
+
+[uuid(436CA5E2-2D50-43C7-9735-E760F299439E), object, pointer_default(unique)]
+interface ICoreWebView2Settings12 : ICoreWebView2Settings11 {
+  /// The `IsNonClientRegionSupportEnabled` property enables web pages to use the 
+  /// `app-region` CSS style. Changing the `IsNonClientRegionSupportEnabled` property
+  /// takes effect after the next navigation of the top level document. Defaults to `FALSE`.
+  /// 
+  /// When this property is `TRUE`, then all the following non-client region support 
+  /// will be enabled:
+  /// 1. Draggable Regions will be enabled and treated as a window's title bar. 
+  /// Draggable Regions are non-client regions on a webpage that are exposed through the css
+  /// attribute `app-region` and can take the values `drag` or `no-drag`. When set to 
+  /// `drag`, these regions will be treated like the window's title bar, supporting 
+  /// dragging of the entire WebView and its host app window, the title bar context menu
+  /// upon right click, and the maximizing to fill the window and restoring the window size
+  /// upon double click of the html element. 
+  ///
+  /// When set to `FALSE`, then all non-client region support will be disabled. Web
+  /// pages will not be able to use the `app-region` CSS style.
+  [propget] HRESULT IsNonClientRegionSupportEnabled([out, retval] BOOL* enabled);
```

Comment:
The name "non-client" is confusing me a lot.

This property is controlling a feature named app-region, and its value named drag, which really should have been named title-bar. None of those is "non-client" :).

It also confuses me because the scenario for this is a WebView in an app window where the non-client area has been turned off. So there _is_ no non-client region in this scenario, rather a title bar that's mimicking the behavior of a non-client area.

I can't find good documentation, and can't figure out if there are other values for app-region. If so I'm not sure if this property is controlling app-region or specifically drag.

Keeping the naming aligned would be good. I also don't think "support" is contributing.

So: `IsAppRegionEnabled`, or `IsAppRegionDragEnabled` (if there are other values of app-region that this doesn't control).

Reply:
Yes, this feature is meant for apps that don't wish to use default Windows non-client areas. It enables apps to be able to implement non-client region behaviors using Webview2 only. The goal is not to eliminate the non-client region behavior or nomenclature, but rather to give apps more control over how they behave. For all intents and purposes the term "non-client region" is very much still appropriate here to describe the overall purpose of this API. 

We do in fact plan to add support for other non-client regions, including minimize/maximize, close,  resize etc... This settings API sets the foundation for being able to control when to use all of these features.